core/src/banking_stage/consumer.rs

```diff
@@ -1804,70 +1804,62 @@ mod tests {
     fn test_consume_buffered_packets_interrupted() {
```

Comment:
imo this test is in a much better state after this:

if we receive to few batches due to a bug in the `Consumer`, we won't hang indefinitely anymore. We will fail with the new count check.